pkg/plugins/runtime/k8s/webhooks/resourceadmissionchecker.go

```diff
@@ -0,0 +1,127 @@
+package webhooks
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slices"
+	v1 "k8s.io/api/admission/v1"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/config/core"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/version"
+)
+
+type ResourceAdmissionChecker struct {
```

Comment:
nit: I'd strick to Validator suffix, but it's up to you

Reply:
That was my initial choice, but then I discovered there is already a thing called [AdmissionValidator](https://github.com/kumahq/kuma/blob/2962747ee1e4350a23e1ec0581d4f78132ac9dcb/pkg/plugins/common/k8s/composite_validator.go#L11) and since I'm not implementing the interface probably it's better not to use `AdmissionValidator` so it's less confusing